modules/get_me_some_ice_cream/states/conversation_flow.py

```diff
@@ -0,0 +1,33 @@
+import ravestate as rs
+import ravestate_interloc as interloc
+import ravestate_rawio as rawio
+
+from get_me_some_ice_cream.states.detection import prop_flavor, prop_scoops
+
+prop_flavor_scoop_tuple_list = rs.Property(name="flavor_scoop_tuple_list", default_value=[], allow_write=True,
+                                           allow_read=True, always_signal_changed=True)
+
+
+@rs.state(
+    cond=interloc.prop_all.pushed().detached().min_age(2),
+    write=rawio.prop_out)
+def prompt_order(ctx: rs.ContextWrapper):
+    ctx[rawio.prop_out] = "Want some ice cream?"
+
+
+@rs.state(
+    cond=prop_flavor.changed() | prop_scoops.changed(),
+    write=rawio.prop_out)
+def check_scoops_flavor_combined(ctx: rs.ContextWrapper):
+    if prop_flavor.read() is not None and prop_scoops.read() is not None:
+        ctx[rawio.prop_out] = "I heard you ordered {} scoops of the flavor {}".format(prop_scoops.read(),
+                                                                                      prop_flavor.read())
+        prop_flavor_scoop_tuple_list.write(prop_flavor_scoop_tuple_list.read() + [(prop_flavor.read(), prop_scoops.read())])
```

Comment:
another possibility would be to use `prop_flavor_scoop_tuple_list.push()`, using it as a parent property instead of a list. But this is purely syntactic sugar 🍦